src/Paramore.Brighter.MessagingGateway.Kafka/KafkaMessageProducer.cs

```diff
@@ -192,12 +194,62 @@ public void Init()
         /// <exception cref="ChannelFailureException">The Kafka client  has entered an unrecoverable state</exception>
         public void Send(Message message)
         {
+            SendWithDelay(message, TimeSpan.Zero);
+        }
+
+        /// <summary>
+        /// Sends the specified message.
+        /// </summary>
+        /// <remarks>
+        ///  Usage of the Kafka async producer is much slower than the sync producer. This is because the async producer
+        /// produces a single message and waits for the result before producing the next message. By contrast the synchronous
+        /// producer queues work and uses a dedicated thread to dispatch
+        /// </remarks>
+        /// <param name="message">The message.</param>
+        /// <param name="cancellationToken">Allows cancellation of the in-flight send operation</param>
+        public async Task SendAsync(Message message, CancellationToken cancellationToken = default)
+        {
+            await SendWithDelayAsync(message, TimeSpan.Zero, cancellationToken);
+        }
+        
+        /// <summary>
+        /// Sends the message with the given delay
+        /// </summary>
+        /// <remarks>
+        /// No delay support implemented
+        /// </remarks>
+        /// <param name="message">The message to send</param>
+        /// <param name="delay">The delay to use</param>
+        public void SendWithDelay(Message message, TimeSpan? delay = null)
+        {
+            delay ??= TimeSpan.Zero;
+            if (delay != TimeSpan.Zero)
+            {
+                if (Scheduler is IAmAMessageSchedulerSync sync)
+                {
+                    sync.Schedule(message, delay.Value);
+                    return;
+                }
+                  
+                if (Scheduler is IAmAMessageSchedulerAsync async)
```

Comment:
We can default back to sync context, but the alternative is to assume you have the right scheduler. Again avoid async as a name